src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -77,9 +87,54 @@ public JsonParser(TypeToken<T> type) {
         this.type = type;
         this.gson = new GsonBuilder()
                 .registerTypeAdapter(Date.class, new DateAdapter().nullSafe())
+                .registerTypeAdapter(StartTime.class,
+                        new JobStartTimeJsonAdapter().nullSafe())
                 .registerTypeAdapter(Stats.class, new StatsDeserializer())
                 .registerTypeAdapter(Arguments.class, new ArgumentsDeserializer())
                 .create();
+        this.startJobDateFormat = JsonParser.START_TIME_FORMAT;
+    }
+
+    /**
+     * Creates a new JsonParser for the given type. Server's time zone is
+     * specified in the @{link ClientConfig}.
+     *
+     * @param type A TypeToken describing the type this parser produces.
+     * @param config ClientConfig used to fetch master's time zone
+     */
+    private JsonParser(TypeToken<T> type, ClientConfig config) {
+        this.type = type;
+        this.gson = new GsonBuilder()
+                .registerTypeAdapter(Date.class, new DateAdapter().nullSafe())
+                .registerTypeAdapter(StartTime.class,
+                        new JobStartTimeJsonAdapter().nullSafe())
+                .registerTypeAdapter(Stats.class, new StatsDeserializer())
+                .registerTypeAdapter(Arguments.class, new ArgumentsDeserializer())
+                .create();
+
+        synchronized (START_TIME_FORMAT) {
+            this.startJobDateFormat = (SimpleDateFormat) START_TIME_FORMAT.clone();
```

Comment:
Java `clone` should generally be avoided unless you know how to handle the demons this black magic might summon into this world.
Just create a helper method that takes all the relevant parts and puts them together into a new `SimpleDateFormat`.